process/block/gasConsumption.md

```diff
@@ -0,0 +1,115 @@
+## Gas consumption implementation
+
+### 1. High level overview
+
+A new gas computation component is needed throughout the block creation mechanism, during transactions selection.
+
+The main purpose of this component will be to help filling the block efficiently, in respect with the gas limits defined in config.
+
+First, incoming mini blocks referenced by metachain will be added. Each mini block will be verified against `MaxGasLimitPerMiniBlock` then the new component will check if each mini block can be added in the initial limit of 50% of `MaxGasLimitPerBlock`. Once the limits are reached, the component will store the remaining mini blocks as pending, until outgoing transactions are handled.
+
+Second, the outgoing list of transactions will be checked against `MaxGasLimitPerTx`, then against the remaining 50% of `MaxGasLimitPerBlock`. If there is not enough "space" for all transactions, the component will first verify if there are any pending mini blocks from the above step. If none, it will continue to add transactions. Otherwise, it will return the last index added.
+
+If the outgoing transactions didn't fill the entire 50% space allocated for them, this component will automatically try to add more pending mini blocks if any, until the block is filled.
+
+Throughout the selection process, up until the next explicit reset called, gasComputation component will store the consumed gas.
+
+Another feature that must be satisfied is the option to decrease the limits when needed. Decreasing limits and resetting them should be done separately.
+
+### 2. Proposed interface
+
+```go
+type GasComputation interface {
+    CheckIncomingMiniBlocks(
+        miniBlocks   []data.MiniBlockHeaderHandler,
+        transactions []data.TransactionHandler,
```

Comment:
do you need also the txHash for each tx here so you can match with the miniBlockHeaderHandler without re-computing the hash?